src/openloco/windows/townwnd.cpp

```diff
@@ -1,31 +1,791 @@
+            window->min_height = 161;
+            window->max_width = 600;
+            window->max_height = 440;
+            window->flags |= window_flags::resizable;
+
+            auto skin = objectmgr::get<interface_skin_object>();
+            if (skin != nullptr)
+            {
+                window->colours[0] = skin->colour_0B;
+                window->colours[1] = skin->colour_0C;
+            }
+            // 0x00499C0D end
+
+            window->var_848 = -1;
+        }
+
+        // TODO(avgeffen): only needs to be called once.
+        common::initEvents();
+
+        window->current_tab = 0;
+        window->invalidate();
+
+        window->widgets = town::widgets;
+        window->enabled_widgets = town::enabledWidgets;
+        window->holdable_widgets = 0;
+        window->event_handlers = &town::events;
+        window->activated_widgets = 0;
+        window->disabled_widgets = 0;
+        window->init_scroll_widgets();
+        town::initViewport(window);
+
+        return window;
+    }
+
+    namespace population
     {
-        registers regs;
-        regs.dx = townId;
-        call(0x00499B7E, regs);
-        return (window*)regs.esi;
+        static widget_t widgets[] = {
+            commonWidgets(223, 161, string_ids::title_town_population),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x00499469
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+        }
+
+        // 0x004994F9
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            gfx::drawpixelinfo_t* clipped = nullptr;
+            if (!gfx::clip_drawpixelinfo(&clipped, dpi, self->x, self->y + 44, self->width, self->height - 44))
+                return;
+
+            auto town = townmgr::get(self->number);
+            *(int32_t*)&*commonFormatArgs = town->history_min_population;
+
+            // Draw Y label and grid lines.
+            for (int16_t yPos = self->height - 57; yPos > 14; yPos -= 20)
+            {
+                const uint16_t xPos = 39;
+                gfx::fill_rect(clipped, xPos, yPos, xPos + 241, yPos, colour::get_shade(self->colours[1], 4));
+
+                gfx::draw_string_494C78(*clipped, xPos, yPos - 6, colour::black, string_ids::population_graph_people, &*commonFormatArgs);
+
+                *(int32_t*)&*commonFormatArgs += 1000;
+            }
+
+            int8_t month = current_month();
+            int16_t year = current_year();
+            int8_t yearSkip = 0;
+
+            for (uint8_t i = town->history_size - 1; i > 0; i--)
+            {
+                const uint16_t xPos = 41 + i;
+                const uint16_t yPos = 2;
+
+                // Draw horizontal year and vertical grid lines.
+                if (month == 0)
+                {
+                    if (yearSkip == 0)
```

Comment:
`if ((year %3) == (current_year() % 3)` (with the latter part stored in a variable) would remove the need for a `yearSkip` variable.

Reply:
That requires doing two divisions though, so I understand why Sawyer did it as-is…